lib/index.js

```diff
@@ -372,6 +394,28 @@ function opt (option) {
   }
 };
 
+/* chan and rmchan sockopt methods. only relevant for subscription sockets */
+Socket.prototype.chan       = function(){
+  var args = arguments;
+  var channels = args.length;
+  while(channels--) {
+    if (nn.Chan(this.binding, nn.NN_SUB_SUBSCRIBE, args[channels]) != -1){
+      this.channels[args[channels]] = true;
+    }
+  };
+}
+
+Socket.prototype.rmchan     = function(){
+  var args = arguments;
+  var channels = args.length;
+  while(channels--) {
+    if (this.channels[args[channels]] &&
+      nn.Chan(this.binding, nn.NN_SUB_UNSUBSCRIBE, args[channels]) != -1) {
+        this.channels[args[channels]] = false;
```

Comment:
Maybe we should be deleting the key from the obj, rather than setting it false?  Eh, I'm on the fence.  Thoughts?